File: frontend/src/modules/purchase-orders/types/purchase-order.ts
export type OrdenCompraEstado = 'borrador' | 'enviada' | 'parcialmente_recibida' | 'completada' | 'cancelada'

export interface OrdenCompraDetalle {
  id: number
  orden_id: number
  producto_id: number
  cantidad: number
  precio_unitario: number
  subtotal: number
  producto_nombre?: string
}

export interface DetalleItem extends OrdenCompraDetalleCreate {
  id?: number
  orden_id?: number
  producto_nombre?: string
  subtotal: number
}

export interface OrdenCompra {
  id: number
  numero: string
  proveedor_id: number
  proveedor_nombre?: string
  usuario_id: number
  estado: OrdenCompraEstado
  observaciones: string | null
  total: number
  created_at: string
  updated_at: string
  detalles: OrdenCompraDetalle[]
}

export interface OrdenCompraFilter {
  estado?: OrdenCompraEstado
}

export interface OrdenCompraDetalleCreate {
  producto_id: number
  cantidad: number
  precio_unitario: number
}

export interface OrdenCompraCreate {
  proveedor_id: number
  usuario_id: number
  observaciones?: string | null
  estado?: OrdenCompraEstado
  detalles: OrdenCompraDetalleCreate[]
}

export interface OrdenCompraUpdate {
  observaciones?: string | null
  estado?: OrdenCompraEstado
}

export const ESTADOS_INMODIFICABLES = ['completada', 'cancelada'] as const

File: backend/app/schemas/purchase_order.py
from datetime import datetime
from enum import Enum
from pydantic import BaseModel, ConfigDict


class OrdenCompraEstado(str, Enum):
    borrador = 'borrador'
    enviada = 'enviada'
    parcialmente_recibida = 'parcialmente_recibida'
    completada = 'completada'
    cancelada = 'cancelada'


class OrdenCompraDetalleCreate(BaseModel):
    producto_id: int
    cantidad: int
    precio_unitario: int


class OrdenCompraDetalleResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    orden_id: int
    producto_id: int
    cantidad: int
    precio_unitario: int
    subtotal: int


class OrdenCompraCreate(BaseModel):
    proveedor_id: int
    usuario_id: int
    observaciones: str | None = None
    estado: OrdenCompraEstado = OrdenCompraEstado.borrador
    detalles: list[OrdenCompraDetalleCreate]


class OrdenCompraUpdate(BaseModel):
    observaciones: str | None = None
    estado: OrdenCompraEstado | None = None


class OrdenCompraResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    numero: str
    proveedor_id: int
    proveedor_nombre: str | None = None
    usuario_id: int
    estado: OrdenCompraEstado
    observaciones: str | None
    total: int
    created_at: datetime
    updated_at: datetime
    detalles: list[OrdenCompraDetalleResponse] = []
